### PR TITLE
refactor(api): wrap all errors FTW

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

### CODE REVIEW

File: engine/api/application/application_delete.go [hunk 1]
```diff
@@ -7,29 +7,26 @@ import (
...
-		return err
+		return sdk.WrapError(err, "DeleteApplication> Cannot delete application variable")
 	}
 
 	// Delete groups
 	query := `DELETE FROM application_group WHERE application_id = $1`
 	if _, err := db.Exec(query, applicationID); err != nil {
-		log.Warning("DeleteApplication> Cannot delete application gorup: %s\n", err)
-		return err
+		return sdk.WrapError(err, "DeleteApplication> Cannot delete application gorup")
```
Comment (fsamin) [inline engine/api/application/application_delete.go]: gorup ?

File: engine/api/action/builtin.go [hunk 2]
```diff
@@ -229,8 +229,7 @@ func CreateBuiltinArtifactActions(db *gorp.DbMap) error {
 	if err != nil && err == sql.ErrNoRows {
 		err = createBuiltinArtifactUploadAction(db)
 		if err != nil {
-			log.Warning("CreateBuiltinArtifactActions> CreateBuiltinArtifactActions err:%s", err.Error())
-			return err
+			return sdk.WrapError(err, "CreateBuiltinArtifactActions> CreateBuiltinArtifactActions err:%s", err.Error())
```
Comment (fsamin) [inline engine/api/action/builtin.go]: FTW, it's useless to print the error in the message

File: engine/api/environment.go [hunk 3]
```diff
@@ -243,8 +231,7 @@ func (api *API) updateEnvironmentsHandler() Handler {
 						}
 						err = environment.InsertVariable(tx, env.ID, varEnv, getUser(ctx))
 						if err != nil {
-							log.Warning("updateEnvironments: Cannot insert variable %s:  %s\n", varEnv.Name, err)
-							return err
+							return sdk.WrapError(err, "updateEnvironments: Cannot insert variable %s:  %s", varEnv.Name, err)
```
Comment (fsamin) [inline engine/api/environment.go]: don't print the error

File: engine/api/build_log.go [hunk 4]
```diff
@@ -127,32 +117,28 @@ func (api *API) getBuildLogsHandler() Handler {
...
+			return sdk.WrapError(sdk.ErrForbidden, "getBuildLogsHandler> No enought right on this environment %s: ", envName)
 
 		}
 
 		// Check that pipeline exists
 		p, err := pipeline.LoadPipeline(api.mustDB(), projectKey, pipelineName, false)
 		if err != nil {
-			log.Warning("getBuildLogsHandler> Cannot load pipeline %s: %s\n", pipelineName, err)
-			return sdk.ErrPipelineNotFound
+			return sdk.WrapError(sdk.ErrPipelineNotFound, "getBuildLogsHandler> Cannot load pipeline %s", pipelineName)
```
Comment (sguiheux) [inline engine/api/build_log.go]: sdk.ErrPipelineNotFound => err

File: engine/api/build_log.go [hunk 5]
```diff
@@ -127,32 +117,28 @@ func (api *API) getBuildLogsHandler() Handler {
...
 
 		}
 
 		// Check that application exists
 		a, err := application.LoadByName(api.mustDB(), api.Cache, projectKey, appName, getUser(ctx))
 		if err != nil {
-			log.Warning("getBuildLogsHandler> Cannot load application %s: %s\n", appName, err)
-			return sdk.ErrApplicationNotFound
+			return sdk.WrapError(sdk.ErrApplicationNotFound, "getBuildLogsHandler> Cannot load application %s", appName)
```
Comment (sguiheux) [inline engine/api/build_log.go]: sdk.ErrApplicationNotFound => err

File: engine/api/workflow/execute_node_run.go [hunk 6]
```diff
@@ -141,8 +141,7 @@ func addJobsToQueue(db gorp.SqlExecutor, stage *sdk.Stage, run *sdk.WorkflowNode
 
 	conditionsOK, err := sdk.WorkflowCheckConditions(stage.Conditions(), run.BuildParameters)
 	if err != nil {
-		log.Warning("addJobsToQueue> Cannot compute prerequisites on stage %s(%d): err", stage.Name, stage.ID, err)
-		return err
+		return sdk.WrapError(err, "addJobsToQueue> Cannot compute prerequisites on stage %s(%d): err", stage.Name, stage.ID, err)
```
Comment (fsamin) [inline engine/api/workflow/execute_node_run.go]: err

File: engine/api/workflow_queue.go [hunk 7]
```diff
@@ -26,8 +26,7 @@ func (api *API) postWorkflowJobRequirementsErrorHandler() Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
 		body, err := ioutil.ReadAll(r.Body)
 		if err != nil {
-			log.Warning("requirementsErrorHandler> %s", err)
-			return err
+			return sdk.WrapError(err, "requirementsErrorHandler> %s", err)
```
Comment (fsamin) [inline engine/api/workflow_queue.go]: err

File: engine/api/application_variable.go [hunk 8]
```diff
@@ -41,60 +40,51 @@ func (api *API) restoreAuditHandler() Handler {
...
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "restoreAuditHandler: Cannot start transaction ")
 		}
 		defer tx.Rollback()
 
 		err = application.DeleteAllVariable(tx, app.ID)
 		if err != nil {
-			log.Warning("restoreAuditHandler: Cannot delete variables for application %s:  %s\n", appName, err)
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "restoreAuditHandler: Cannot delete variables for application %s:  %s", appName, err)
```
Comment (fsamin) [inline engine/api/application_variable.go]: err

File: engine/api/application_variable.go [hunk 9]
```diff
@@ -41,60 +40,51 @@ func (api *API) restoreAuditHandler() Handler {
...
-				log.Warning("restoreAuditHandler: Cannot insert variable %s for application %s:  %s\n", v.Name, appName, err)
-				return err
+				return sdk.WrapError(err, "restoreAuditHandler: Cannot insert variable %s for application %s", v.Name, appName)
 			}
 		}
 
 		if err := tx.Commit(); err != nil {
-			log.Warning("restoreAuditHandler: Cannot commit transaction:  %s\n", err)
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "restoreAuditHandler: Cannot commit transaction:  %s", err)
```
Comment (fsamin) [inline engine/api/application_variable.go]: err

File: engine/api/pipeline_parameter.go [hunk 10]
```diff
@@ -103,19 +94,16 @@ func (api *API) updateParametersInPipelineHandler() Handler {
...
-			log.Warning("updateParametersInPipelineHandler> Cannot GetAllParametersInPipeline: %s\n", err)
-			return err
+			return sdk.WrapError(err, "updateParametersInPipelineHandler> Cannot GetAllParametersInPipeline")
 		}
 
 		tx, err := api.mustDB().Begin()
 		if err != nil {
-			log.Warning("updateParametersInPipelineHandler: Cannot start transaction: %s", err)
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "updateParametersInPipelineHandler: Cannot start transaction: %s", err)
```
Comment (sguiheux) [inline engine/api/pipeline_parameter.go]: sdk.ErrUnknownError > err

File: engine/api/pipeline_parameter.go [hunk 11]
```diff
@@ -191,8 +175,7 @@ func (api *API) updateParametersInPipelineHandler() Handler {
 		}
 
 		if err := tx.Commit(); err != nil {
-			log.Warning("updateParametersInPipelineHandler: Cannot commit transaction: %s", err)
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "updateParametersInPipelineHandler: Cannot commit transaction: %s", err)
```
Comment (sguiheux) [inline engine/api/pipeline_parameter.go]: ->> err

File: engine/api/project_variable.go [hunk 12]
```diff
@@ -38,63 +37,54 @@ func (api *API) restoreProjectVariableAuditHandler() Handler {
...
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "restoreProjectVariableAuditHandler: Cannot start transaction ")
 
 		}
 		defer tx.Rollback()
 
 		if err := project.DeleteAllVariable(tx, p.ID); err != nil {
-			log.Warning("restoreProjectVariableAuditHandler: Cannot delete variables for project %s:  %s\n", key, err)
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "restoreProjectVariableAuditHandler: Cannot delete variables for project %s:  %s", key, err)
```
Comment (sguiheux) [inline engine/api/project_variable.go]: err

File: engine/api/project_variable.go [hunk 13]
```diff
@@ -115,8 +105,7 @@ func (api *API) getVariablesInProjectHandler() Handler {
 
 		p, err := project.Load(api.mustDB(), api.Cache, key, getUser(ctx), project.LoadOptions.WithVariables)
 		if err != nil {
-			log.Warning("deleteVariableFromProject: Cannot load %s: %s\n", key, err)
-			return sdk.ErrNotFound
+			return sdk.WrapError(sdk.ErrNotFound, "deleteVariableFromProject: Cannot load %s", key)
```
Comment (sguiheux) [inline engine/api/project_variable.go]: >> err

File: engine/api/project_variable.go [hunk 14]
```diff
@@ -181,15 +165,13 @@ func (api *API) updateVariablesInProjectHandler() Handler {
...
-			return sdk.ErrNotFound
+			return sdk.WrapError(sdk.ErrNotFound, "updateVariablesInProjectHandler: Cannot load %s", key)
 
 		}
 
 		tx, err := api.mustDB().Begin()
 		if err != nil {
-			log.Warning("updateVariablesInProjectHandler: Cannot start transaction: %s\n", err)
-			return sdk.ErrNotFound
+			return sdk.WrapError(sdk.ErrNotFound, "updateVariablesInProjectHandler: Cannot start transaction")
```
Comment (sguiheux) [inline engine/api/project_variable.go]: ->> err

File: engine/api/project_variable.go [hunk 15]
```diff
@@ -198,14 +180,12 @@ func (api *API) updateVariablesInProjectHandler() Handler {
...
-			log.Warning("updateVariablesInProjectHandler: Cannot preload variables values: %s\n", err)
-			return err
+			return sdk.WrapError(err, "updateVariablesInProjectHandler: Cannot preload variables values")
 
 		}
 
 		if err := project.DeleteAllVariable(tx, p.ID); err != nil {
-			log.Warning("updateVariablesInProjectHandler: Cannot delete all variables for project %s: %s\n", p.Key, err)
-			return sdk.ErrNotFound
+			return sdk.WrapError(sdk.ErrNotFound, "updateVariablesInProjectHandler: Cannot delete all variables for project %s", p.Key)
```
Comment (sguiheux) [inline engine/api/project_variable.go]: >>> err

File: engine/api/project_variable.go [hunk 16]
```diff
@@ -243,30 +221,26 @@ func (api *API) updateVariablesInProjectHandler() Handler {
...
 		if err := project.UpdateLastModified(tx, api.Cache, getUser(ctx), p); err != nil {
-			log.Warning("updateVariablesInProjectHandler: Cannot update last modified:  %s\n", err)
-			return err
+			return sdk.WrapError(err, "updateVariablesInProjectHandler: Cannot update last modified")
 		}
 
 		if err := tx.Commit(); err != nil {
-			log.Warning("updateVariablesInProjectHandler: Cannot commit transaction: %s\n", err)
-			return sdk.ErrNotFound
+			return sdk.WrapError(sdk.ErrNotFound, "updateVariablesInProjectHandler: Cannot commit transaction")
```
Comment (sguiheux) [inline engine/api/project_variable.go]: >>>err

File: engine/api/router_util.go [hunk 17]
```diff
@@ -27,8 +27,7 @@ func deleteUserPermissionCache(ctx context.Context, store cache.Store) {
 func WriteJSON(w http.ResponseWriter, r *http.Request, data interface{}, status int) error {
 	b, e := json.Marshal(data)
 	if e != nil {
-		log.Warning("WriteJSON> unable to marshal : %s", e)
-		return sdk.ErrUnknownError
+		return sdk.WrapError(sdk.ErrUnknownError, "WriteJSON> unable to marshal : %s", e)
```
Comment (sguiheux) [inline engine/api/router_util.go]: > wrongRequest

File: engine/api/scheduler.go [hunk 18]
```diff
@@ -29,23 +29,20 @@ func (api *API) getSchedulerApplicationPipelineHandler() Handler {
...
-			return errP
+			return sdk.WrapError(errP, "getSchedulerApplicationPipelineHandler> Cannot load pipeline %s", pipelineName)
 
 		}
 
 		//Load environment
 		if err := r.ParseForm(); err != nil {
-			log.Warning("getSchedulerApplicationPipelineHandler> Cannot parse form: %s\n", err)
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "getSchedulerApplicationPipelineHandler> Cannot parse form")
```
Comment (sguiheux) [inline engine/api/scheduler.go]: ->> wrongrequest

File: engine/api/scheduler.go [hunk 19]
```diff
@@ -93,23 +88,20 @@ func (api *API) addSchedulerApplicationPipelineHandler() Handler {
...
-			return errP
+			return sdk.WrapError(errP, "addSchedulerApplicationPipelineHandler> Cannot load pipeline %s", pipelineName)
 
 		}
 
 		//Load environment
 		if err := r.ParseForm(); err != nil {
-			log.Warning("getSchedulerApplicationPipelineHandler> Cannot parse form: %s\n", err)
-			return sdk.ErrUnknownError
+			return sdk.WrapError(sdk.ErrUnknownError, "getSchedulerApplicationPipelineHandler> Cannot parse form")
```
Comment (sguiheux) [inline engine/api/scheduler.go]: -> wrongrequest

File: engine/api/stage.go [hunk 20]
```diff
@@ -85,8 +79,7 @@ func (api *API) getStageHandler() Handler {
 
 		stageID, err := strconv.ParseInt(stageIDString, 10, 60)
 		if err != nil {
-			log.Warning("getStageHandler> Stage ID must be an int: %s", err)
-			return sdk.ErrWrongRequest
+			return sdk.WrapError(sdk.ErrWrongRequest, "getStageHandler> Stage ID must be an int: %s", err)
```
Comment (sguiheux) [inline engine/api/stage.go]: InvalidID

**Pull request merged**